File: frontend/app/src/main/java/com/example/commov/viewmodel/SettingsViewModel.java
package com.example.commov.viewmodel;

import android.content.Context;

import com.example.commov.data.local.LocaleHelper;

public final class SettingsViewModel {
    public SettingsUiState getState(Context context) {
        return new SettingsUiState(LocaleHelper.getSavedLanguage(context));
    }

    public boolean changeLanguage(Context context, String language) {
        if (LocaleHelper.getSavedLanguage(context).equals(language)) {
            return false;
        }

        LocaleHelper.setLanguage(context, language);
        return true;
    }
}

File: frontend/app/src/main/java/com/example/commov/viewmodel/SettingsViewModel.kt
package com.example.commov.viewmodel

import android.content.Context
import com.example.commov.data.local.LocaleHelper

class SettingsViewModel {
    fun getState(context: Context): SettingsUiState {
        return SettingsUiState(LocaleHelper.getSavedLanguage(context))
    }

    fun changeLanguage(context: Context, language: String): Boolean {
        if (LocaleHelper.getSavedLanguage(context) == language) {
            return false
        }

        LocaleHelper.setLanguage(context, language)
        return true
    }
}
